app/models/aggregated_cash_income.rb

```diff
@@ -0,0 +1,85 @@
+class AggregatedCashIncome
+  include ActiveModel::Model
+
+  CATEGORIES= %i[
+      benefits
+      friends_or_family
+      maintenance_in
+      property_or_lodger
+      pension
+  ].freeze
+
+  def self.category_attributes(categories)
+    categories.each do |category|
+      attr_accessor "check_box_#{category}".to_sym
+      attr_accessor "#{category}_month_1".to_sym
+      attr_accessor "#{category}_month_2".to_sym
+      attr_accessor "#{category}_month_3".to_sym
+    end
+  end
+
+  attr_accessor :legal_aid_application_id, :amount, :transaction_date
+
+  category_attributes CATEGORIES
+
+  validate :checkbox_integrity
+
+  def checkbox_integrity
+    CATEGORIES.each do |category|
+      if "check_box_#{category}".present?
+        errors.add(":#{category}_month_1", "Cannot be empty") unless ":#{category}_month_1".present?
+        errors.add(":#{category}_month_2", "Cannot be empty") unless ":#{category}_month_2".present?
+        errors.add(":#{category}_month_3", "Cannot be empty") unless ":#{category}_month_3".present?
+      end
+    end
+  end
+
+  def update!(params)
+    params.each do |key, value|
+      __send__("#{key}=",value)
+    end
+    save_cash_transaction_records
+  end
+
+  private
+
+  def save_cash_transaction_records
+    CATEGORIES.each do |category|
+      save_category(category) if checkbox_for(category) == true
+    end
+  end
+
+  def save_category(category)
+    # for each monthly amount attribute we need a transaction to be created
+    # each transaction needs to have a different calculated date based on the
+    # numbered suffix - _month_1 is the previous month
+    # need to pass the attr_accessor into calculated date
+    transaction_type = TransactionType.credits.find_by(name: category)
+    CashTransaction.create!(legal_aid_application_id: legal_aid_application_id,
+                            transaction_type_id: transaction_type.id,
+                            transaction_date: calculated_date,
+                            amount:  __send__(:amount))
```

Comment:
This `CashTransaction.create!` statement needs to be wrapped in a `(1..3).each do |i|` loop and the variable `i` needs to be passed in to the `calculated_date` method.